src/lib.rs

```diff
@@ -717,6 +717,325 @@ where
     }
 }
 
+macro_rules! impl_primitive_ops_ratio {
+    (Ratio<T> for $primitive:ident) => {
+        impl<T> Add<Ratio<T>> for $primitive
+        where
+            T: Clone + Integer + ::core::convert::From<$primitive>,
+        {
+            type Output = Ratio<T>;
+
+            #[inline]
+            fn add(self, rhs: Ratio<T>) -> Ratio<T> {
+                Ratio::new(rhs.numer + T::from(self), rhs.denom)
```

Comment:
As noted above, you lose the bigint performance win if you force `T::from` here.  But this implementation is also incorrect.  Look at the `arith_impl!` macro -- when adding an integer to a ratio, you need to multiply it by the denominator!  e.g. your code would compute 1/3 + 1 = 2/3, but it should be 4/3.